launcher/agent/agent_test.go

```diff
@@ -338,6 +344,170 @@ func TestFetchContainerImageSignatures(t *testing.T) {
 	}
 }
 
+type returnVal struct {
+	result []oci.Signature
+	err    error
+}
+
+type failingClient struct {
+	results  map[string][]returnVal
+	numTimes map[string]int
+}
+
+func NewFailingClient(mymap map[string][]returnVal) signaturediscovery.Fetcher {
+	numTimes := map[string]int{}
+	for k := range mymap {
+		numTimes[k] = 0
+	}
+	return &failingClient{
+		results:  mymap,
+		numTimes: numTimes,
+	}
+}
+
+func (f *failingClient) FetchImageSignatures(_ context.Context, targetRepository string) ([]oci.Signature, error) {
+	attempt := f.numTimes[targetRepository]
+	r := f.results[targetRepository][intMin(attempt, len(f.results[targetRepository])-1)]
+	f.numTimes[targetRepository] = attempt + 1
+	return r.result, r.err
+}
```

Comment:
nit: Looks like signatures are returned in a round robin fashion, could you add some comments to this method?

Alternatively, you might consider changing the implementation to something like w/o calling `intMin`
```
func (f *failingClient) FetchImageSignatures(_ context.Context, targetRepository string) ([]oci.Signature, error) {
	attempt := f.numTimes[targetRepository]
	r := f.results[targetRepository][attempt]
	f.numTimes[targetRepository] = (attempt + 1)  % len(f.results[targetRepository])
	return r.result, r.err
}
```

Reply:
Yeah fair, that makes more sense. Made the change.

Reply:
Only change I am going to make is not modulo, as its not exactly round robin. Design here is to return the last value forever. So basically I'm just taking the nicer setting of the value from your implementation :)